### PR TITLE
Nevergrad support

### PR DESCRIPTION
**Pull request by lucasplagwitz**
Features

- nevergrad support with tests and exmple (switch not implemented yet. Design choice..) 
- CNN optimzied exmple to examples/neural_network/
- content based tests for warnings (remove counter)
- enable more skopt input params

Bugs

- remove probabilites from tested_config_lists
- adjustments for learning curves: seaborn version 0.11 no longer supports lineplot with previous syntax. Should be backwards compatible, only tested for seaborn 0.10.1.
- Float labels could not save to MongoDB (dot not allowed as key in dict) -> replace every dot in class distribution keys.
- disable multiprocessing for preprocessing elements in optimum_pipe (example: NeuroBranch in Preprocessing)

Other

- rename: save_memory -> decrease_memory
- add comments
- add tests 
- pep8 changes
- increase skopt runs 10 -> 25 (10 is the initial-(random)-run count)
- small updates for keras wrapper: remove reload_weights (functionality no longer exists), no distinguish between tensorflow.kers.model and keras.model
- add xlrd to dependencies
- allow estimator_type in [None, 'transformer'] for element without predict method (transformer)

### CODE REVIEW

File: test/processing_tests/test_outer_folds.py [hunk 1]
```diff
@@ -183,6 +183,8 @@ def test_save_predictions_and_feature_importances(self):
                     self.assertTrue(np.sum(len(fold.feature_importances) == 7))
             else:
                 self.assertTrue(np.sum(len(fold.validation.y_pred) for fold in config.inner_folds) == 0)
+                self.assertTrue(np.sum(len(fold.validation.probabilities) for fold in config.inner_folds) == 0)
```
Comment (RLeenings) [inline test/processing_tests/test_outer_folds.py:186]: ist das immer eine Liste? Weil wenn es None ist, dann würde das failen.

**Review by RLeenings — APPROVED** [review]
